src/api/src/factories/TargetUserDefinesFactory.ts

```diff
@@ -1,931 +1,125 @@
-      // performance options
-      UserDefine.Boolean(UserDefineKey.NO_SYNC_ON_ARM),
-      UserDefine.Enum(
-        UserDefineKey.ARM_CHANNEL,
-        ['AUX1', 'AUX2', 'AUX3', 'AUX4', 'AUX5', 'AUX6'],
-        'AUX1'
-      ),
-      UserDefine.Boolean(UserDefineKey.FEATURE_OPENTX_SYNC, true),
-      UserDefine.Boolean(UserDefineKey.USE_500HZ),
-      // compat
-      UserDefine.Boolean(UserDefineKey.UART_INVERTED, true),
-      // other
-      UserDefine.Boolean(UserDefineKey.USE_DYNAMIC_POWER),
-    ];
-
-    const FM30_RX: UserDefine[] = [
-      // regulatory options
-      UserDefine.Boolean(UserDefineKey.REGULATORY_DOMAIN_ISM_2400, true),
-      // binding
-      UserDefine.Text(UserDefineKey.BINDING_PHRASE, '', true),
-      // hybrid switches
-      UserDefine.Boolean(UserDefineKey.HYBRID_SWITCHES_8, true),
-      UserDefine.Boolean(UserDefineKey.ENABLE_TELEMETRY),
-      // performance options
-      UserDefine.Boolean(UserDefineKey.NO_SYNC_ON_ARM),
-      UserDefine.Boolean(UserDefineKey.LOCK_ON_FIRST_CONNECTION, true),
-      UserDefine.Boolean(UserDefineKey.USE_500HZ),
-      UserDefine.Boolean(UserDefineKey.USE_DIVERSITY, true),
-    ];
-
-    const GHOST_ATTO_2400_RX: UserDefine[] = [
-      // regulatory options
-      UserDefine.Boolean(UserDefineKey.REGULATORY_DOMAIN_ISM_2400, true),
-      // binding
-      UserDefine.Text(UserDefineKey.BINDING_PHRASE, '', true),
-      // hybrid switches
-      UserDefine.Boolean(UserDefineKey.HYBRID_SWITCHES_8, true),
-      UserDefine.Boolean(UserDefineKey.ENABLE_TELEMETRY),
-      // performance options
-      UserDefine.Boolean(UserDefineKey.NO_SYNC_ON_ARM),
-      UserDefine.Boolean(UserDefineKey.LOCK_ON_FIRST_CONNECTION, true),
-      UserDefine.Boolean(UserDefineKey.USE_500HZ),
-    ];
-
-    const DIY_2400_RX_ESP8285_SX1280: UserDefine[] = [
-      // regulatory options
-      UserDefine.Boolean(UserDefineKey.REGULATORY_DOMAIN_ISM_2400, true),
-      // binding
-      UserDefine.Text(UserDefineKey.BINDING_PHRASE, '', true),
-      // hybrid switches
-      UserDefine.Boolean(UserDefineKey.HYBRID_SWITCHES_8, true),
-      UserDefine.Boolean(UserDefineKey.ENABLE_TELEMETRY),
-      // performance options
-      UserDefine.Boolean(UserDefineKey.NO_SYNC_ON_ARM),
-      UserDefine.Boolean(UserDefineKey.LOCK_ON_FIRST_CONNECTION, true),
-      UserDefine.Boolean(UserDefineKey.USE_500HZ),
-      // other options
-      UserDefine.Boolean(UserDefineKey.USE_DIVERSITY, false),
-      // network
-      UserDefine.Boolean(UserDefineKey.AUTO_WIFI_ON_BOOT),
-      UserDefine.Text(UserDefineKey.AUTO_WIFI_ON_INTERVAL, '20', true),
-      UserDefine.Text(UserDefineKey.HOME_WIFI_SSID),
-      UserDefine.Text(UserDefineKey.HOME_WIFI_PASSWORD),
-    ];
-
-    const DIY_2400_RX_ESP8285_SX1280_NO_DIVERSITY: UserDefine[] = DIY_2400_RX_ESP8285_SX1280.filter(
-      (item) => item.key !== UserDefineKey.USE_DIVERSITY
+  build(target: string, deviceService: DeviceService): UserDefine[] {
```

Comment:
I think it would be better to mark this factory class with `@Service()` annotation and inject `DeviceService` using constructor based injection. Then you will not need to pass DeviceService manually upstream.